### PR TITLE
Externalise config

### PR DESCRIPTION
**Pull request by phunkyfish**
v3.14.0
- Added: Externalised season/episode and genre config to allow users support other formats/languages
- Added: Server OpenWebIf version now reported by addon

### CODE REVIEW

File: src/client.cpp [hunk 1]
```diff
@@ -217,7 +217,7 @@ const char *GetBackendName(void)
 
 const char *GetBackendVersion(void)
 {
-  static const char *strBackendVersion = "UNKNOWN";
+  static const char *strBackendVersion = enigma ? enigma->GetServerVersion() : "unknown";
```
Comment (ksooo) [inline src/client.cpp]: Shouldn't "unkown" be localisable?

File: src/enigma2/Admin.cpp [hunk 2]
```diff
@@ -62,82 +68,77 @@ bool Admin::GetDeviceInfo()
...
+  Logger::Log(LEVEL_NOTICE, "%s - E2DeviceName: %s", __FUNCTION__, serverName.c_str());
+
+  m_deviceInfo = DeviceInfo(serverName, enigmaVersion, imageVersion, webIfVersion, webIfVersionAsNum);
+  Settings::GetInstance().SetDeviceInfo(m_deviceInfo);
 
   return true;
 }
 
-unsigned int Admin::GetWebIfVersion(std::string versionString)
+unsigned int Admin::ParseWebIfVersion(std::string &webIfVersion)
```
Comment (ksooo) [inline src/enigma2/Admin.cpp]: `const std::string &webifversion` ?

File: src/enigma2/extract/ShowInfoExtractor.cpp [hunk 3]
```diff
@@ -1,27 +1,38 @@
...
+  //       GET_EPISODE_PATTERN));
+  // m_episodeSeasonPatterns.emplace_back(
+  //     EpisodeSeasonPattern(MASTER_YEAR_EPISODE_PATTERN, 
+  //       GET_EPISODE_PATTERN));
+  // m_episodeSeasonPatterns.emplace_back(
+  //     EpisodeSeasonPattern(MASTER_EPISODE_NO_PREFIX_PATTERN, 
+  //       GET_EPISODE_NO_PREFIX_PATTERN));
+
+  // m_yearPatterns.emplace_back(std::regex(GET_YEAR_PATTERN));
+  // m_yearPatterns.emplace_back(std::regex(GET_YEAR_EPISODE_PATTERN));
```
Comment (ksooo) [inline src/enigma2/extract/ShowInfoExtractor.cpp]: do you want to keep this?

File: pvr.vuplus/addon.xml.in [hunk 4]
```diff
@@ -71,7 +71,10 @@
     <description lang="de_DE">VU+ -Oberfläche; Unterstützt Live TV &amp; Aufnahmen, EPG und Timer.</description>
     <description lang="el_GR">Frontend για το VU+. Υποστηρίζει ροές Live TV &amp; Εγγραφές, EPG, Χρονοδιακόπτες.</description>
     <description lang="en_AU">VU+ frontend; supporting streaming of Live TV &amp; Recordings, EPG, Timers.</description>
-    <description lang="en_GB">Enigma2 frontend; supporting streaming of Live TV &amp; Recordings, EPG, Timers, Autotimers.</description>
+    <description lang="en_GB">Enigma2 frontend - supporting streaming of Live TV &amp; Recordings, EPG, Timers, Autotimers.
+    
+For documentation visit: https://github.com/kodi-pvr/pvr.vuplus/blob/master/README.md
```
Comment (ksooo) [inline pvr.vuplus/addon.xml.in]: This looks strange. Is the "hard" formatting intended?

**Comment by ksooo** [comment]
Just had a very brief look at the changes, could not find anything oviously wrong. If you think it's okay, go ahead.

File: pvr.vuplus/addon.xml.in (hunk 4 above)
Comment (phunkyfish) [reply pvr.vuplus/addon.xml.in]: Yes it is. To get the text to appear at the left the line needs to start as above.

File: src/enigma2/extract/ShowInfoExtractor.cpp (hunk 3 above)
Comment (phunkyfish) [reply src/enigma2/extract/ShowInfoExtractor.cpp]: Missed this, apologies

File: src/enigma2/Admin.cpp (hunk 2 above)
Comment (phunkyfish) [reply src/enigma2/Admin.cpp]: Doh, there is always one.

File: src/client.cpp (hunk 1 above)
Comment (phunkyfish) [reply src/client.cpp]: I actually have several of these in my next PR, there are quite a few. This is one of them.

**Comment by phunkyfish** [comment]
Ok, I’ll merge this once home and and the fix ups are in.

**Pull request merged**